src/main/java/org/truffleruby/language/arguments/RubyArguments.java

```diff
@@ -80,11 +94,38 @@
         packed[ArgumentIndicies.SELF.ordinal()] = self;
         packed[ArgumentIndicies.BLOCK.ordinal()] = block;
 
+        // TODO: We need to somehow set it to "flattened" or something else
+        //   if it's flattened.
+        packed[ArgumentIndicies.KW_ARGS_OPTIMIZABLE_FLAG.ordinal()] = null; // Set to null for now
+
+        // Copy arguments into `packed`, with the correct `packed` size
         ArrayUtils.arraycopy(arguments, 0, packed, RUNTIME_ARGUMENT_COUNT, arguments.length);
 
         return packed;
     }
 
+    private static Object[] flattenArguments(RubyHash arguments) {
+        // There are 2 types of Hashes: Packed hashes, Long hashes
+        // Packed hashes does not have `firstInSequence`, while long hashes do
+        if (arguments.firstInSequence != null) {
```

Comment:
There's helpers for all this in `HashOperations` and the hash 'strategies' - `PackedHashStrategy` etc.